pages/index.vue

```diff
@@ -147,15 +165,35 @@ export default defineComponent({
      * @todo 1. Send the request to update the task to the backend server.
      * @todo 2. Update the task in the dom.
      */
-    updateTask(_index, _id) {},
+    updateTask(_index, _id) {
+      if (!this.newTitle) return
+      const headers = {
+        Authorization: 'Token ' + this.$store.getters.token,
+      }
+      const data = {
+        title: this.newTitle,
+      }
+      this.$axios.put(`/todo/${_id}/`, data, { headers }).then(() => {
+        this.todos[_index].editing = !this.todos[_index].editing
+        this.todos[_index].title = this.newTitle
+        this.newTitle = ''
+        this.prev = -1
+      })
+    },
     /**
      * toggle visibility of input and buttons for a single todo
      * @argument {number} index - index of element to toggle
      * @todo add in bindings in dom so that 'hideme' class is dynamic or use conditional rendering
      * @hint read about class bindings in vue
      */
     editTask(index) {
+      this.$toast.info('Press Done to Save the Update before Proceeding!')
+      this.newTitle = ''
+      if (this.prev !== -1) {
+        this.todos[this.prev].editing = !this.todos[this.prev].editing
+      }
```

Comment:
you had the access to index in this function could have done things without prev too